pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -185,14 +222,43 @@ func (c *podIpMetricsCollector) setupDirectoryWatcher(dir string) error {
 
 		for {
 			select {
-			case _, ok := <-watcher.Events:
+			case e, ok := <-watcher.Events:
 				if !ok {
 					glog.Error("watcher is not ok")
 					return
 				}
 				if err := c.listIpAddresses(dir); err != nil {
 					return
 				}
+				names := strings.Split(e.Name, "/")
+				f := names[len(names)-1]
```

Comment:
We should ensure that this file is an IPv4 address. We do not want to update the map for an IPv6 address otherwise it will grow till 4 billion entries (theoretically). We also want to skip this for other files.

Reply:
Done.